### PR TITLE
feat: publish ongoing game data

### PR DESCRIPTION
**Pull request by ze97286**
Closes #11209

what's missing: deprecate existing API - which will be done separately.

### CODE REVIEW

File: protos/sources/vega/events/v1/events.proto [hunk 1]
```diff
@@ -1136,6 +1136,41 @@ message TeamMemberStats {
...
+  string score = 2;
+  // The current staking balance of the party. Only populated if the game has a requirement for it.
+  string staking_balance = 3;
+  // The current open volume of the party. Only populated if the game has a requirement for it.
+  string open_volume = 4;
+  // The current traded volume of the party. Only populated if the game has a requirement for it.
+  string trade_volume = 5;
+  // Total fees paid by the party during the relevant period.
+  string total_fees_paid = 6;
+  // Is the party eligible for reward for this game based on the current information.
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto]:
```suggestion
  // If the party is eligible for a reward for this game based on the current information. 
```

File: protos/sources/vega/events/v1/events.proto [hunk 2]
```diff
@@ -1136,6 +1136,41 @@ message TeamMemberStats {
...
+  string total_fees_paid = 6;
+  // Is the party eligible for reward for this game based on the current information. 
+  bool is_eligible = 7;
+  // If the party is a memeber of a team, this is their relative position in the sorting order of the team. 
+  // Empty if not a team, or if the party is not eligible. 
+  optional uint64 rank = 8;
+}
+
+message GameTeamScore {
+  string team = 1;
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:1161]:
```suggestion
 // Team ID
  string team = 1;
```

File: protos/sources/vega/events/v1/events.proto [hunk 3]
```diff
@@ -1136,6 +1136,41 @@ message TeamMemberStats {
...
+  string score = 2;
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:1162]:
```suggestion
 // Team's current collective score in the game.
  string score = 2;
```

File: protos/sources/vega/events/v1/events.proto [hunk 4]
```diff
@@ -1355,6 +1390,9 @@ enum BusEventType {
   // Event containing the IDs of orders cancelled by a party on a market.
   BUS_EVENT_TYPE_CANCELLED_ORDERS = 89;
 
+  // Event containing the near real time game scores for parties and teams.
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:1393]:
```suggestion
  // Event containing the near realtime game scores for parties and teams.
```

File: protos/sources/vega/events/v1/events.proto [hunk 5]
```diff
@@ -1136,6 +1136,41 @@ message TeamMemberStats {
...
+}
+
+message GameTeamScore {
+  string team = 1;
+  string score = 2;
+  repeated GamePartyScore party_scores = 3;
+}
+
+message GameScores {
+  string game_id = 1;
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:1167]:
```suggestion
 // Game ID.
  string game_id = 1;
```

File: protos/sources/vega/events/v1/events.proto [hunk 6]
```diff
@@ -1136,6 +1136,41 @@ message TeamMemberStats {
...
+  int64 time = 2;
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:1168]:
```suggestion
 // Current time in RFC3339Nano.
  int64 time = 2;
```

File: protos/sources/vega/events/v1/events.proto [hunk 7]
```diff
@@ -1136,6 +1136,41 @@ message TeamMemberStats {
...
+  string epoch = 3;
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:1169]:
```suggestion
 // Current epoch.
  string epoch = 3;
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 8]
```diff
@@ -103,6 +103,20 @@ service TradingDataService {
     option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Orders"};
   }
 
+  // List the current scores of teams in games
+  //
+  // Get a list of team scores for the given filters
+  rpc ListGameTeamScore(ListGameTeamScoreRequest) returns (ListGameTeamScoreResponse) {
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Team Game Scores"};
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:110]: We don't really need a whole new category for this API:
```suggestion
    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Games"};
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 9]
```diff
@@ -103,6 +103,20 @@ service TradingDataService {
...
+  }
+
+  // List the current scores of parties in games
+  //
+  // Get a list of party scores for the given filters
+  rpc ListGamePartyScore(ListGamePartyScoreRequest) returns (ListGamePartyScoreResponse) {
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Party Game Scores"};
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:117]:
```suggestion
    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Games"};
```

File: protos/sources/vega/events/v1/events.proto [hunk 10]
```diff
@@ -1136,6 +1136,51 @@ message TeamMemberStats {
...
+message GamePartyScore {
+  // Game ID.
+  string game_id = 1;
+  // The party ID.
+  string party = 2;
+  // optional team ID.
+  optional string team_id = 3;
+  // Current epoch.
+  int64 epoch = 4;
+  // Current time in RFC3339Nano.
```
Comment (wwestgarth) [inline protos/sources/vega/events/v1/events.proto:1149]:
```suggestion
  // Timestamp in Unix nanoseconds when these scores were generated.
```

File: protos/sources/vega/events/v1/events.proto [hunk 11]
```diff
@@ -1136,6 +1136,51 @@ message TeamMemberStats {
...
+}
+
+message GameTeamScore {
+  // Game ID.
+  string game_id = 1;
+  // Team ID
+  string team_id = 2;
+  // Current epoch.
+  int64 epoch = 3;
+  // Current time in RFC3339Nano.
```
Comment (wwestgarth) [inline protos/sources/vega/events/v1/events.proto:1173]:
```suggestion
  // Timestamp in Unix nanoseconds when these scores were generated.
```

File: protos/sources/vega/events/v1/events.proto [hunk 12]
```diff
@@ -1136,6 +1136,51 @@ message TeamMemberStats {
...
+  // Empty if not a team, or if the party is not eligible.
+  optional uint64 rank = 11;
+}
+
+message GameTeamScore {
+  // Game ID.
+  string game_id = 1;
+  // Team ID
+  string team_id = 2;
+  // Current epoch.
```
Comment (wwestgarth) [inline protos/sources/vega/events/v1/events.proto:1171]:
```suggestion
  // Epoch when these scores were generated.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 13]
```diff
@@ -103,6 +103,20 @@ service TradingDataService {
     option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Orders"};
   }
 
+  // List the current scores of teams in games
+  //
+  // Get a list of team scores for the given filters
+  rpc ListGameTeamScore(ListGameTeamScoreRequest) returns (ListGameTeamScoreResponse) {
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:109]: Given its a `List` API is should be pluralised no? `ListGameTeamScores`?

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 14]
```diff
@@ -103,6 +103,20 @@ service TradingDataService {
     option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Orders"};
   }
 
+  // List the current scores of teams in games
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:106]:
```suggestion
  // List game team scores
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 15]
```diff
@@ -103,6 +103,20 @@ service TradingDataService {
...
+  //
+  // Get a list of team scores for the given filters
+  rpc ListGameTeamScore(ListGameTeamScoreRequest) returns (ListGameTeamScoreResponse) {
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Team Game Scores"};
+  }
+
+  // List the current scores of parties in games
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:113]:
```suggestion
  // List game party scores
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 16]
```diff
@@ -1316,6 +1330,80 @@ message GetStopOrderResponse {
...
+  // Optional pagination control.
+  optional Pagination pagination = 1;
+  // Party games filter filter contains all filtering conditions and values that are applied to the party games score listing.
+  optional GamePartyScoresFilter filter = 2;
+}
+
+message GamePartyScoresFilter {
+  // Restrict game scores to those in the given games.
+  repeated string game_ids = 1;
+  // Restrict game scores to those in the given teams.
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:1380]:
```suggestion
  // Restrict game scores to the parties in the given teams.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 17]
```diff
@@ -1316,6 +1330,80 @@ message GetStopOrderResponse {
...
+  repeated string team_ids = 2;
+  // Restrict game scores to those given parties.
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:1382]:
```suggestion
  // Restrict game scores to those relating to the given parties.
```

File: protos/sources/vega/events/v1/events.proto [hunk 18]
```diff
@@ -1136,6 +1136,51 @@ message TeamMemberStats {
   string notional_volume = 2;
 }
 
+// Scores per party per game with eligibility context.
+message GamePartyScore {
+  // Game ID.
+  string game_id = 1;
+  // The party ID.
+  string party = 2;
+  // optional team ID.
```
Comment (wwestgarth) [inline protos/sources/vega/events/v1/events.proto:1145]: // Team the party belongs to. Unset if the party is not part of a team.

File: datanode/gateway/graphql/schema.graphql [hunk 19]
```diff
@@ -1120,6 +1120,22 @@ type Query {
...
 
+  "Get a list of current game team scores. If provided, the filter will be applied to the list of games/teams to restrict the results."
+  gameTeamScores(
+    "Optional filter to restrict the results of the list."
+    filter: GameTeamScoreFilter
+    "Optional pagination information"
+    pagination: Pagination
+  ): GameTeamScoreConnection
+
+  "Get a list of current game party orders. If provided, the filter will be applied to the list of games/parties to restrict the results."
```
Comment (peterbarrow) [inline datanode/gateway/graphql/schema.graphql:1131]:
```suggestion
  "Get a list of current game party scores. If provided, the filter will be applied to the list of games/parties to restrict the results."
```

**Review by peterbarrow — DISMISSED** [review]

**Review by peterbarrow — DISMISSED** [review]

File: datanode/gateway/graphql/schema.graphql [hunk 20]
```diff
@@ -6001,6 +6033,39 @@ type StopOrderConnection {
   pageInfo: PageInfo
 }
 
+"Connection type for retrieving cursory-based paginated game team score information"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6036]:
```suggestion
"Connection type for retrieving cursor-based paginated game team score information"
```

File: datanode/gateway/graphql/schema.graphql [hunk 21]
```diff
@@ -6001,6 +6033,39 @@ type StopOrderConnection {
...
+type GameTeamScoreConnection {
+  "The game team scores in this connection"
+  edges: [GameTeamScoreEdge!]
+  "The pagination information"
+  pageInfo: PageInfo
+}
+
+"Connection type for retrieving cursory-based paginated game party score information"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6044]:
```suggestion
"Connection type for retrieving cursor-based paginated game party score information"
```

File: datanode/gateway/graphql/schema.graphql [hunk 22]
```diff
@@ -7351,3 +7416,44 @@ type TimeWeightedNotionalPosition {
...
+  score: String!
+  "The staking balance of the party in the game, will be populated only if the game has a requirement for it"
+  stakingBalance: String
+  "The open volume of the party in the game, will be populated only if the game has a requirement for it"
+  openVolume: String 
+  "The total fees paid by the party in the game during the relevant period"
+  totalFeesPaid: String!
+  "Is the party eligible for a reward in this game based on existing information"
+  isEligible: Boolean!
+  "If the party is a memeber of a team, this is their relative position in the sorting order of the team's scores"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:7456]:
```suggestion
  "If the party is a member of a team, this is their relative position in the sorting order of the team's scores"
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 23]
```diff
@@ -1316,6 +1330,80 @@ message GetStopOrderResponse {
   vega.events.v1.StopOrderEvent order = 1;
 }
 
+// Request that is sent when executing a query for a list of team game scores.
+message ListGameTeamScoresRequest {
+  // Optional pagination control.
+  optional Pagination pagination = 1;
+  // Team games filter filter contains all filtering conditions and values that are applied to the team games score listing.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1337]:
```suggestion
  // Team games filter contains all filtering conditions and values that are applied to the team games score listing.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 24]
```diff
@@ -1316,6 +1330,80 @@ message GetStopOrderResponse {
...
+  vega.events.v1.GameTeamScore node = 1;
+  // Cursor that can be used to fetch further pages.
+  string cursor = 2;
+}
+
+// Request that is sent when executing a query for a list of party game scores.
+message ListGamePartyScoresRequest {
+  // Optional pagination control.
+  optional Pagination pagination = 1;
+  // Party games filter filter contains all filtering conditions and values that are applied to the party games score listing.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1373]:
```suggestion
  // Party games filter contains all filtering conditions and values that are applied to the party games score listing.
```

**Review by EVODelavega — DISMISSED** [review]

**Review by EVODelavega — DISMISSED** [review]

**Review by cdummett — DISMISSED** [review]

File: datanode/gateway/graphql/schema.graphql [hunk 25]
```diff
@@ -5969,6 +5985,22 @@ input StopOrderFilter {
   liveOnly: Boolean
 }
 
+input GameTeamScoreFilter {
+  "Zero or more game IDs to filter by"
+  game_ids: [ID!]
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:5990]: Should this be formatted as ‘gameIds’? Same question for team IDs fields

Comment (ze97286) [reply datanode/gateway/graphql/schema.graphql:5990]: good catch!

**Review by cdummett — APPROVED** [review]

**Review by candida-d — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**